common/djangoapps/third_party_auth/signals/tests/test_handlers.py

```diff
@@ -115,3 +117,205 @@ def test_saml_config_signal_handlers(
                     f"Expected '{error_message}' in error message for {description}, "
                     f"got: {error_calls[0][1][1]}"
                 )
+
+
+@ddt.ddt
+class TestSAMLProviderConfigUpdates(TestCase):
+    """
+    Test SAML provider config updates based on SAML configuration changes.
+    """
+
+    def setUp(self):
+        # Create test sites
+        self.site1 = Site.objects.get_or_create(domain='test-site1.com', name='Site 1')[0]
```

Comment:
Note: For my proposed ddt test, you might need a `_create_site(site_id)` method instead of these sites in setUp.